File: src/Models/LetsEncryptServer/Account/Requests/PatchAccountRequest.cs
using MaksIT.Core.Abstractions.Webapi;


namespace MaksIT.Models.LetsEncryptServer.Account.Requests;

public class PatchAccountRequest : PatchRequestModelBase {

  public string Description { get; set; }

  public bool? IsDisabled { get; set; }

  public List<string>? Contacts { get; set; }

  public List<PatchHostnameRequest>? Hostnames { get; set; }
}

File: src/MaksIT.WebUI/src/models/certsUI/account/requests/PatchAccountRequest.ts
import { PatchHostnameRequest, PatchHostnameRequestSchema } from './PatchHostnameRequest'
import { PatchRequestModelBase, PatchRequestModelBaseSchema } from '../../../PatchRequestModelBase'
import { array, boolean, object, Schema, string } from 'zod'

export interface PatchAccountRequest extends PatchRequestModelBase {
  description?: string
  isDisabled?: boolean
  contacts?: string[]
  hostnames?: PatchHostnameRequest[]
}

export const PatchAccountRequestSchema: Schema<PatchAccountRequest> = PatchRequestModelBaseSchema.and(
  object({
    description: string().optional(),
    isDisabled: boolean().optional(),
    contacts: array(string()).optional(),
    hostnames: array(PatchHostnameRequestSchema).optional()
  })
)

